### PR TITLE
Support of downgrading database schema

### PR DESCRIPTION
**Pull request by tobiasKaminsky**
Downgrade from 25->24 is not really necessary.

### CODE REVIEW

File: src/main/java/com/owncloud/android/providers/FileContentProvider.java [hunk 1]
```diff
@@ -1152,6 +1153,13 @@ public void onUpgrade(SQLiteDatabase db, int oldVersion, int newVersion) {
                 }
             }
         }
+
+        @Override
+        public void onDowngrade(SQLiteDatabase db, int oldVersion, int newVersion) {
+            if (oldVersion == 25 && newVersion == 24) {
+                // nothing needs to be done as the upgrade was adding columns only if they did not exist
```
Comment (mario) [inline src/main/java/com/owncloud/android/providers/FileContentProvider.java]: Well if we want to be precise, then we could remove the columns xD

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky looking at the code we basically "do nothing" with this code change except adding an empty method, right?

File: src/main/java/com/owncloud/android/providers/FileContentProvider.java (hunk 1 above)
Comment (tobiasKaminsky) [reply src/main/java/com/owncloud/android/providers/FileContentProvider.java]: Schema 25 only re-added the columns if they were missing. This is mainly a fix for some, like Jason.
The columns are needed, e.g. in upgrade to 25 CAPABILITIES_SERVER_SLOGAN is added if it is missing, and this is also added on a fresh installation at https://github.com/nextcloud/android/blob/8272bfab161000b0f55cd8143436dd90dd5f7450/src/main/java/com/owncloud/android/providers/FileContentProvider.java#L1254-L1254

**Comment by tobiasKaminsky** [comment]
@AndyScherzinger from "our" side this is true. But with calling this function the internal scheme db version is set to 24 after the function runs without an exception.
And if we later add new columns/tables we can then easily add them in the same way for downgrade.

**Comment by AndyScherzinger** [comment]
:+1: ah, fine!
But shouldn't we then put in a log statement that we downgraded to 24 and remove the unused constant?

File: src/main/java/com/owncloud/android/providers/FileContentProvider.java (hunk 1 above)
Comment (AndyScherzinger) [reply src/main/java/com/owncloud/android/providers/FileContentProvider.java]: master is on 24 so 25 is already used for e2e (or whatever might be merged before that...)

**Comment by AndyScherzinger** [comment]
pinging @mario for review :)

**Comment by AndyScherzinger** [comment]
rebased ❗️

**Comment by tobiasKaminsky** [comment]
@mario please review, after that I can do a rebase and merge it.

**Comment by AndyScherzinger** [comment]
pinging @mario for review :) This one important (spams the gplay console with crashes like crazy due to e2e users who moved to 2.0.0 stable....)

**Comment by mario** [comment]
👍

**Pull request merged**